lib/LaTeXML/Core/Box.pm

```diff
@@ -236,12 +236,21 @@ sub computeSize {
 #**********************************************************************
 # What about Kern, Glue, Penalty ...
 
+# stubs to avoid "shallow" Fatal errors on broken documents
+sub multiply { my ($self) = @_; return $self; }
+sub subtract { my ($self) = @_; return $self; }
+sub add      { my ($self) = @_; return $self; }
+sub negate   { my ($self) = @_; return $self; }
+sub divide   { my ($self) = @_; return $self; }
+sub getX     { my ($self) = @_; return $self; }
+sub getY     { my ($self) = @_; return $self; }
+sub ptValue  { return; }
```

Comment:
I see it avoiding Fatals, but probably next to impossible to find the problem if we ever try to debug the underlying problem.  Perhaps better to signal an Error (or Warning?) first?

Perhaps many such errors are due to something like "1.1em" getting treated as straight text but should have been dimensions (or numbers or...)

Reply:
Most of the ones I debugged already had both warning and error traces before the Fatal, e.g. "missing package X" followed by "undefined" macro \y, and then "missing argument for macro \z". And it is exactly in cases where interpretation went awry and arguments don't arrive at the right places.

So I'm unsure I see the "getting lost" argument if the Fatal is missing - hitting the "100 errors" fatal is a fine upper bound, and in the cases where we're at fewer errors, an HTML doc can be created. And often reading through the HTML file is more helpful than the error messages themselves, for a TeX author.

Reply:
Sure, in the specific cases you're looking at, yeah there will be lots of other errors. But if you're in a situation with just a few errors, you've hidden an important symptom.  If you want to take that approach, better would be to return Dimension(0) rather than $self (which will likely just generate even more errors to paper over)

Reply:
Could even check the box contents for whether it looks like a number :>....  For another time